x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalyzeAction.java

```diff
@@ -0,0 +1,1013 @@
+                    )
+                );
+                queue.add(verifyBlobTask);
+            }
+
+            for (int i = 0; i < workerCount; i++) {
+                processNextTask();
+            }
+        }
+
+        private void processNextTask() {
+            final VerifyBlobTask thisTask = queue.poll();
+            if (isRunning() == false || thisTask == null) {
+                onWorkerCompletion();
+            } else {
+                logger.trace("processing [{}]", thisTask);
+                // NB although all this is on the SAME thread, the per-blob verification runs on a SNAPSHOT thread so we don't have to worry
+                // about local requests resulting in a stack overflow here
+                final TransportRequestOptions transportRequestOptions = TransportRequestOptions.timeout(
+                    TimeValue.timeValueMillis(timeoutTimeMillis - currentTimeMillisSupplier.getAsLong())
+                );
+                transportService.sendChildRequest(
+                    thisTask.node,
+                    BlobAnalyzeAction.NAME,
+                    thisTask.request,
+                    task,
+                    transportRequestOptions,
+                    new TransportResponseHandler<BlobAnalyzeAction.Response>() {
+                        @Override
+                        public void handleResponse(BlobAnalyzeAction.Response response) {
+                            logger.trace("finished [{}]", thisTask);
+                            expectedBlobs.add(thisTask.request.getBlobName()); // each task cleans up its own mess on failure
+                            if (request.detailed) {
+                                synchronized (responses) {
+                                    responses.add(response);
+                                }
+                            }
+                            summary.add(response);
+                            processNextTask();
+                        }
+
+                        @Override
+                        public void handleException(TransportException exp) {
+                            logger.debug(new ParameterizedMessage("failed [{}]", thisTask), exp);
+                            fail(exp);
+                            onWorkerCompletion();
+                        }
+
+                        @Override
+                        public BlobAnalyzeAction.Response read(StreamInput in) throws IOException {
+                            return new BlobAnalyzeAction.Response(in);
+                        }
+                    }
+                );
+            }
+
+        }
+
+        private BlobContainer getBlobContainer() {
+            return repository.blobStore().blobContainer(new BlobPath().add(blobPath));
+        }
+
+        private void onWorkerCompletion() {
+            if (workerCountdown.countDown()) {
+                transportService.getThreadPool().executor(ThreadPool.Names.SNAPSHOT).execute(() -> {
```

Comment:
No AbstractRunnable here? I see that none of the methods called here should fail, perhaps we can make this more explicit here

Reply:
Ok, wrapped in an `AbstractRunnable` for protection in 12ccaba93a08e2ed5363f2c980ce260e059cf409.